app/src/test/java/com/eis/geoCalendar/database/GenericEventDatabasePoolTest.java

```diff
@@ -0,0 +1,178 @@
+    );
+    private static final TypeToken<GenericEvent<String>> EVENT_TOKEN =
+            new TypeToken<GenericEvent<String>>() {
+            };
+    private static final List<GenericEvent<String>> SOME_EVENTS = new ArrayList<>();
+
+    private EventDatabase<GenericEvent<String>> database;
+
+    @BeforeClass
+    public static void fillList() {
+        SOME_EVENTS.add(SIMPLE_EVENT);
+        SOME_EVENTS.add(ANOTHER_SIMPLE_EVENT);
+    }
+
+    @Before
+    public void setDatabase() {
+        database = EventDatabasePool.getInMemoryInstance(
+                RuntimeEnvironment.systemContext,
+                EVENT_TOKEN,
+                DEFAULT_DB_NAME
+        );
+    }
+
+    /**
+     * Just a facade call to allow actual clearing method to be static, to be used in any other
+     * tests that need it.
+     */
+    @After
+    public void clearInstances() {
+        clearActiveInstances();
+    }
+
+    /**
+     * This method is necessary, due to an issue with MySQL, where static instances of the
+     * database cannot be reset between tests automatically.
+     */
+    public static void clearActiveInstances() {
+        RoomEventDatabaseTest.clearActiveInstances();
+        try {
+            //Getting the field named "activeInstances" (the map with all active databases)
+            Field instance = EventDatabasePool.class.getDeclaredField("activeInstances");
+            //Making it accessible
+            instance.setAccessible(true);
+            //Casting it to a Map
+            Map<?, ?> privateField = (Map<?, ?>) instance.get(null);
+            //Making it empty
+            if (privateField != null)
+                privateField.clear();
+        } catch (Exception e) {
+            e.printStackTrace();
+        }
+    }
+
+    /**
+     * Testing whether the database actually starts empty.
+     */
+    @Test
+    public void startsEmpty() {
+        final int expectedStartCount = 0;
+        assertEquals(expectedStartCount, database.count());
+    }
+
+    /**
+     * Testing single insertion.
+     */
+    @Test
+    public void canAddOne() {
+        final int expectedCount = 1;
+        database.saveEvent(SIMPLE_EVENT);
+        assertEquals(expectedCount, database.count());
+    }
+
+    /**
+     * Testing the insertion is actually correct, through {@link StringDao#contains(String)}.
+     */
+    @Test
+    public void canAddOneAndItsCorrect() {
+        database.saveEvent(SIMPLE_EVENT);
+        assertTrue(database.contains(SIMPLE_EVENT));
+    }
+
+    /**
+     * Testing multiple insertions.
+     */
+    @Test
+    public void canAddMultiple() {
+        database.saveEvents(SOME_EVENTS);
+        final int expectedCount = SOME_EVENTS.size();
+        assertEquals(expectedCount, database.count());
+    }
+
+    /**
+     * Testing multiple insertions are actually correct, through {@link StringDao#getAllEntities()}.
+     */
+    @Test
+    public void canAddMultipleAndTheyAreCorrect() {
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Unit tests should not contain more than 1 assert(s).](https://app.codacy.com/manual/CremaLuca/geoCalendar/pullRequest?prid=4822079)